### PR TITLE
refactor(read-preference): merge missing api from native version

### PR DESCRIPTION
**Pull request by mbroadst**
This updates the ReadPreference in the core module to serve as a
full replacement for the one in the porcelain layer.

NODE-1187

### CODE REVIEW

File: lib/topologies/read_preference.js [hunk 1]
```diff
@@ -54,13 +55,54 @@ var ReadPreference = function(preference, tags, options) {
...
+});
+
+/**
+ * Validate if a mode is legal
+ *
+ * @method
+ * @param {string} mode The string representing the read preference mode.
+ * @return {boolean}
+ */
+ReadPreference.isValid = function(mode) {
```
Comment (daprahamian) [inline lib/topologies/read_preference.js]: Can we use a `Set` here?

File: lib/topologies/read_preference.js [hunk 2]
```diff
@@ -54,13 +55,54 @@ var ReadPreference = function(preference, tags, options) {
   }
 };
 
+// Support the deprecated `preference` property introduced in the porcelain layer
+Object.defineProperty(ReadPreference.prototype, 'preference', {
```
Comment (daprahamian) [inline lib/topologies/read_preference.js]: In the future we should replace this whole thing with a class and a getter here

Comment (mbroadst) [reply lib/topologies/read_preference.js]: agreed, but a bit pressed for time today :)

File: lib/topologies/read_preference.js (hunk 1 above)
Comment (mbroadst) [reply lib/topologies/read_preference.js]: like:
```
const validModes = [
  ReadPreference.PRIMARY, 
  ReadPreference.PRIMARY_PREFERRED,
  ReadPreference.SECONDARY,
  ReadPreference.SECONDARY_PREFERRED,
  ReadPreference.NEAREST,
  true,
  false,
  null
];

ReadPreference.isValid = function(mode) {
  return validModes.has(mode);
}
```
?

Comment (daprahamian) [reply lib/topologies/read_preference.js]: Yeah, either by using the native `Set` (do we get that in Node 4?), or by just doing `validModes.indexOf(x) >=0`

Comment (mbroadst) [reply lib/topologies/read_preference.js]: okay I'm into it

**Pull request merged**